packages/core-bridge/src/lib.rs

```diff
@@ -792,6 +792,13 @@ fn worker_shutdown(mut cx: FunctionContext) -> JsResult<JsUndefined> {
     }
 }
 
+/// Drop a reference to a Client, once all references are dropped, the Client will be closed.
+fn client_close(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+    let client = cx.argument::<BoxedClient>(0)?;
+    drop(client);
+    Ok(cx.undefined())
```

Comment:
I think this is purely performative - do you know this is actually doing what you want? I couldn't immediately find a test for it.

AFAICT looking at Neon docs/src dropping a JsBox does nothing - the actual lifetime is managed in JS and the `Finalize` implementation is called when it's freed.

You could do something like put the client behind an option, and _really_ drop it here, but then all other accesses of the client will have this potential error path where the client is already closed, but that's kinda necessary for this to behave how you really want

Reply:
I was meaning to get back to this, forgot to verify it in the PR.

Reply:
It ended up being way more complicated than initially anticipated but I think it's fine.
Managing these object lifecycles isn't trivial but we have to do it in order to avoid leaving the runtime around, it prevents the process from exiting.